src/Microsoft.AspNetCore.ResponseCaching/Interfaces/IResponseCachingCacheKeyModifier.cs

```diff
@@ -5,13 +5,20 @@
 
 namespace Microsoft.AspNetCore.ResponseCaching
 {
-    public interface IResponseCachingCacheKeySuffixProvider
+    public interface IResponseCachingCacheKeyModifier
     {
         /// <summary>
+        /// Create a key segment that is prepended to the default cache key.
+        /// </summary>
+        /// <param name="httpContext">The <see cref="HttpContext"/>.</param>
+        /// <returns>The key segment that will be prepended to the default cache key.</returns>
+        string CreatKeyPrefix(HttpContext httpContext);
```

Comment:
I don't see why we need both of these. The key is never re-parsed, so it doesn't matter to us if the customization is at the start or end. The request for a prefix was for shared cache disambiguation, correct? If that's the case then the suffix data could just as easily be put at the beginning too.